src/Nri/Ui/Block/V6.elm

```diff
@@ -553,23 +553,34 @@ toMark config { backgroundColor, borderColor } =
         Just
             { name = config.label
             , startStyles =
-                [ Css.borderLeft3 borderWidth borderStyle notBottomBorderColor
+                [ Css.borderLeft3 borderWidth borderStyle borderColor_
                 , Css.paddingLeft (Css.px 2)
                 ]
             , styles =
                 [ Css.paddingTop topBottomSpace
                 , Css.paddingBottom topBottomSpace
                 , Css.backgroundColor backgroundColor
-                , Css.borderTop3 borderWidth borderStyle notBottomBorderColor
-                , Css.borderBottom3 borderWidth borderStyle borderColor
+                , Css.borderTop3 borderWidth borderStyle borderColor_
+                , Css.borderBottom3 borderWidth borderStyle borderColor_
                 , MediaQuery.highContrastMode
                     [ Css.property "background-color" "Mark"
                     , Css.property "color" "MarkText"
                     , Css.property "forced-color-adjust" "none"
                     ]
+                , Css.batch
+                    (case config.borderStyle of
+                        Dashed ->
+                            []
+
+                        Underline ->
+                            [ Css.textDecoration Css.underline
+                            , Css.property "text-decoration-thickness" "1px"
+                            , Css.property "text-underline-offset" "4.5px"
```

Comment:
Perhaps 0.2rem or something like that instead? so that this style works well for 30px fonts and 18px fonts.

4.5px looks a bit far to me for an 18px font:

<img width="419" alt="Screenshot 2024-01-12 at 6 39 04 PM" src="https://github.com/NoRedInk/noredink-ui/assets/8811312/9a0c799a-ccb5-40ae-bf69-e8031c2b3f81">

Reply:
The main issue I was running into was getting it aligned with a blank underline, which is why I was moving it at all. Happy to go toward a different solution, I wasn't attached or happy with how this was turning out.

Reply:
Okay! I think I'm following. So border-bottom is used for adding a "blank" underline and text-decoration is used for adding underlines to actual text?

I think this brings me back to my question from yesterday: "Should non-blanks ever be underlined?" Like, is the intention of the underline just to be a different styling for a blank? If so, then I don't think we'd ever need text-decoration at all? (Although I think an API change could clarify the expected behavior a bit.)

Reply:
> Like, is the intention of the underline just to be a different styling for a blank?

Yes. In our context, students thought the dashed bordered rectangle versions of the blanks meant that it was interactive

Reply:
How about this?

| 30px | 18px |
|------|------|
|<img width="723" alt="image" src="https://github.com/NoRedInk/noredink-ui/assets/32229300/a7d5cf4a-23ee-43e1-8eeb-9430f2179cdc">|<img width="443" alt="image" src="https://github.com/NoRedInk/noredink-ui/assets/32229300/f338352b-a1d7-4011-be67-be880cd917ab">|

Instead of adjusting the border offset for the text, I adjusted the line height of the blank, bringing it up rather than down. I also set the width to 0.2rem, which at a default zoom is rendering at 1.5px.

Reply:
I guess I'm still not clear on the advantage of allowing the text to be underlined? I don't think there's a time when we would want to underline text in this way. 🤔

Reply:
Oops, missed the first part of Tessa's question. We don't underline text in Short Response. I'm not 💯 sure but don't think there's a use case for it in scaffolding or Guided Tutorials either? (the other 2 places currently using Block)

Reply:
I was just aiming for some consistency between the two modes. Taking away the names the component is currently using, we have two display modes that we want to be able to choose between:

- Show blanks with dashes
- Show blanks with underlines

Originally, I felt it made more sense to make that a global thing: show _everything_ with dashes, or show _everything_ as underlined. If we are feeling this is not the case, I can scope that down to just blanks.

Reply:
Personally, I think we should not show everything as underlines. The case that really clinches it for me is a blank within an emphasis, which is the one you've got screenshots of up above in this thread:

<img width="688" alt="image" src="https://github.com/NoRedInk/noredink-ui/assets/8811312/46e57dfd-7522-4ff7-81cd-987ec80c8d8f">

I would expect the "underline" style to look like this:
<img width="679" alt="image" src="https://github.com/NoRedInk/noredink-ui/assets/8811312/ddb6bbd8-f46d-43f9-86b5-30c20f4f4cdd">

Not like this:

<img width="674" alt="image" src="https://github.com/NoRedInk/noredink-ui/assets/8811312/26cbf488-c709-46a1-909f-feb7129b10ee">

Reply:
Ah, I get where the other underlines came from. We want to apply underline to just blanks. Or apply to everything, but have the border go all the way around except in blanks, where it should just be on the bottom.

We don't currently have a use case to change the dashes on the other Block elements. The border is there for added contrast in addition to the background color.

Reply:
Fine with me! Here's the update (which looks just like Tessa's example):

<img width="693" alt="image" src="https://github.com/NoRedInk/noredink-ui/assets/32229300/d3fb9e5b-3b71-4123-893a-56a3eebd6ba7">